src/front/glsl/parser.rs

```diff
@@ -342,7 +350,16 @@ impl<'source, 'program, 'options> Parser<'source, 'program, 'options> {
                 Ok(())
             }
             // TODO: handle Shared?
-            _ => Err(ErrorKind::InvalidToken(token)),
+            _ => Err(ErrorKind::InvalidToken(
+                token,
+                vec![
+                    "identifier".into(),
+                    "push_constant".into(),
+                    "std140".into(),
+                    "std430".into(),
+                    "early_fragment_tests".into(),
```

Comment:
We should only use identifier here since unknown layout directives is another error